drivers/ec_host_cmd_periph/Kconfig

```diff
@@ -31,6 +31,55 @@ config EC_HOST_CMD_PERIPH_ESPI
 	  Enable support for Embedded Controller host commands using
 	  the eSPI bus.
 
+config EC_HOST_CMD_PERIPH_SHI
+	bool "Host commands support using SHI"
+	help
+	  Enable support for Embedded Controller host commands using
+	  the Serial Host Interface.
+
 endchoice
 
+if EC_HOST_CMD_PERIPH_SHI
+
+choice
+	prompt "SHI driver"
+	default EC_HOST_CMD_PERIPH_SHI_NPCX if SOC_FAMILY_NPCX
+	default EC_HOST_CMD_PERIPH_SHI_ITE if SOC_IT8XXX2
+
+config EC_HOST_CMD_PERIPH_SHI_NPCX
+	bool "SHI by Nuvoton"
+	depends on DT_HAS_NUVOTON_NPCX_SHI_ENABLED
+	help
+	  This option enables the driver for SHI peripheral in the
+	  Nuvoton NPCX chip.
+
+config EC_HOST_CMD_PERIPH_SHI_ITE
+	bool "SHI by ITE"
+	depends on DT_HAS_ITE_IT8XXX2_SHI_ENABLED
+	help
+	  This option enables the driver for SHI peripheral in the
+	  ITE IT8xxx2 chips family.
+
+endchoice
+
+config EC_HOST_CMD_PERIPH_SHI_MAX_REQUEST
+	hex "Max data size for the version 3 request packet"
+	default 0x220 if EC_HOST_CMD_PERIPH_SHI_NPCX
+	default 0x100 if EC_HOST_CMD_PERIPH_SHI_ITE
```

Comment:
In npcx series, no matter input or output buffer size is 128 bytes. EC_HOST_CMD_PERIPH_SHI_MAX_REQUES is not related to hardware capability directly. But it does influence the performance of flash write command during updating RW firmware. (Please refer [shi.c](https://chromium.googlesource.com/chromiumos/platform/ec/+/refs/heads/main/chip/npcx/shi.c#122) for more detail)

Reply:
I am not aware of any hardware limitations to this.

Reply:
Thanks @MulinChao for clarification.
@fabiobaltieri : it can't be a dts property, since this has to be exported to EC code. There is a host command that must send the response size value to the AP. Extracting it from DTS won't be good, since this function can be shared between eSPI and SHI.

Reply:
I think you could still keep the Kconfig option and change the default based on a dt property, seems like it's mainly used for clock frequency around the code base, for example:  https://github.com/zephyrproject-rtos/zephyr/blob/7c9d7922116c9ce7fded4213f3bbbbbbbbe1c382/soc/arm/atmel_sam0/saml21/Kconfig.defconfig.series#L28-L29

Thinking is if it's a property of the SHI device node, it may be handy to keep all of them together in the dt rather than having one in the driver Kconfig. That said finding the node like is kinda messy. Meh, not sure, up to you :-)

Reply:
As we talked directly, the Kconfig for the maximum request and response sizes have to be exported outside the kernel to the application, so it must be visible independently of the device tree. Also the Kconfig doesn't allow to set the value of DT property minus some const (padding), so there is no good way to set value basing on it. I will just export the fifo size to DTS, leaving this Kconfig as independent value.